src/inference/src/os/win/win_system_conf.cpp

```diff
@@ -77,65 +84,73 @@ void parse_processor_info_win(const char* base_ptr,
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_CORE_ID] = _cores;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_TYPE] = HYPER_THREADING_PROC;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_GROUP_ID] = group;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_GROUP_ID] = group;
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                proc_info[CPU_MAP_CORE_TYPE] = HYPER_THREADING_PROC;
+                proc_info[CPU_MAP_GROUP_ID] = group;
+                _cpu_mapping_table.push_back(proc_info);
+
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[1] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                proc_info[CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
+                proc_info[CPU_MAP_GROUP_ID] = group;
+                _cpu_mapping_table.push_back(proc_info);
 
                 _proc_type_table[0][MAIN_CORE_PROC]++;
                 _proc_type_table[0][HYPER_THREADING_PROC]++;
                 group++;
 
             } else {
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                if ((_processors > group_start) && (_processors <= group_end)) {
+                    proc_info[CPU_MAP_CORE_TYPE] = group_type;
+                    proc_info[CPU_MAP_GROUP_ID] = group_id;
+                    _proc_type_table[0][group_type]++;
+                }
+                _cpu_mapping_table.push_back(proc_info);
             }
             _proc_type_table[0][ALL_PROC] += list_len;
-            proc_count += list_len;
+            _processors += list_len;
             _cores++;
 
         } else if ((info->Relationship == RelationCache) && (info->Cache.Level == 2)) {
             MaskToList(info->Cache.GroupMask.Mask);
 
             if (4 == list_len) {
-                for (int m = 0; m < list_len; m++) {
+                if (_processors < list[list_len - 1] + base_proc) {
```

Comment:
It is '<' or '<='?

Reply:
it is `<`. `=` part has been covered in following for loop.